amethyst_renderer/src/renderer.rs

```diff
@@ -316,3 +316,41 @@ fn init_backend(wb: WindowBuilder, el: &EventsLoop, config: &DisplayConfig) -> R
 
     Ok(Backend(dev, fac, main_target, win))
 }
+
+/// Creates the OpenGL backend.
+#[cfg(all(feature = "opengl", target_os = "macos"))]
+fn init_backend(wb: WindowBuilder, el: &mut EventsLoop, config: &DisplayConfig) -> Result<Backend> {
+    use gfx_window_glutin as win;
+    use glutin::{self, GlContext, GlProfile, GlRequest};
+
+    let ctx = glutin::ContextBuilder::new()
+        .with_multisampling(config.multisampling)
+        .with_vsync(config.vsync)
+        .with_gl_profile(GlProfile::Core)
+        .with_gl(GlRequest::Latest);
+
+    let (win, dev, fac, color, depth) = win::init::<ColorFormat, DepthFormat>(wb, ctx, el);
+    let size = win.get_inner_size().ok_or(Error::WindowDestroyed)?.into();
+    let main_target = Target::new(
+        ColorBuffer {
+            as_input: None,
+            as_output: color,
+        },
+        DepthBuffer {
+            as_input: None,
+            as_output: depth,
+        },
+        size,
+    );
+
+    // On Mac 10.14 we need to resize the window after creation
+    // this is related to this issue amethyst/amethyst#972
+    el.poll_events(|_| {});
```

Comment:
Maybe we could annotate this more so that we don't forget to remove it when the bug is fixed?

Reply:
@Moxinilian does a TODO work or a follow-up issue?

Reply:
I think an issue might be better, but at the same time if it ends up at the bottom of the pile...

(I wonder if we have old critical issues we forgot about)

Reply:
Ive added a fixme and ill create an issue here shortly.